File: Practice/11/C++/involution/involution.cpp
#include <iostream>
using namespace std;
int main() {
	setlocale(LC_ALL, "Russian");
	double number, degree, result;
	cout << "Введите число" << endl;
	cin >> number;
	cout << "Введите степень" << endl;
	cin >> degree; 
	result = number;
	if (degree == 1) {
		cout << result;
	}
	else if (degree > 1) {
		while (degree > 1) {
			result = result * number;
			degree = degree - 1;
		}
		cout << result << endl;
	}
	if (degree == 0) {
		cout << "1" << endl;
	}
	else if (degree < 0) {
		while (degree < -1) {
			result = result * number;
			degree = degree + 1;
		}
		cout << 1 / result;
	}
}

File: Practice/11/Python/involution/involution.py
print ("Введите число")
number = int(input())
print("Введите степень")
degree = int(input())
result = number;
if ((degree == 1) and (number == 0)):
	print ("1")
if degree == 1:
	print(result)
if degree > 1:
		while degree > 1:
			result = result * number
			degree = degree - 1
		print (result)
if degree == 0: 

	print("1")
elif degree < 0:
		while degree < -1: 
			result = result * number
			degree = degree + 1
		print(1 / result)